### PR TITLE
feat(handlers/guild): better return type for getAuditLogs()

### PR DESCRIPTION
**Pull request by itohatweb**

### CODE REVIEW

File: src/types/guild.ts [hunk 1]
```diff
@@ -380,6 +381,90 @@ export interface GuildEmbed {
...
+/** https://discord.com/developers/docs/resources/audit-log#audit-log-object */
+export interface AuditLogPayload {
+  /** list of webhooks found in the audit log */
+  webhooks: WebhookPayload[];
+  /** list of users found in the audit log */
+  users: UserPayload[];
+  /** list of audit log entries */
+  audit_log_entries: AuditLogEntryPayload[];
+  /** list of partial integration objects */
+  integrations: Partial<Integration>[];
```
Comment (ayntee) [inline src/types/guild.ts:393]:
```suggestion
  integrations: Pick<Integration, "id" | "name" | "type" | "account">[];
```

File: src/types/guild.ts [hunk 2]
```diff
@@ -380,6 +381,90 @@ export interface GuildEmbed {
...
+  /** changes made to the target_id */
+  changes?: AuditLogChangePayload[];
+  /** the user who made the changes */
+  user_id: string;
+  /** id of the entry */
+  id: string;
+  /** type of action that occured */
+  action_type: AuditLogs;
+  /** additional info for certain action types */
+  options?: OptionalAuditEntryInfoParam;
```
Comment (ayntee) [inline src/types/guild.ts:409]:
```suggestion
  options?: OptionalAuditEntryInfoPayload;
```

File: src/types/guild.ts [hunk 3]
```diff
@@ -380,6 +381,90 @@ export interface GuildEmbed {
...
+  /** the reason for the change (0-512 characters) */
+  reason?: string;
+}
+
+/** https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object-optional-audit-entry-info */
+export interface OptionalAuditEntryInfoParam {
```
Comment (ayntee) [inline src/types/guild.ts:415]:
```suggestion
export interface OptionalAuditEntryInfoPayload {
```

File: src/types/guild.ts (hunk 1 above)
Comment (itohatweb) [reply src/types/guild.ts:393]: Are you sure that only these parameters exist? And there is no chance of the others be there too?

Comment (ayntee) [reply src/types/guild.ts:393]: > Are you sure that only these parameters exist? And there is no chance of the others be there too?

Yes and not sure about the second question, maybe you should ask in API server,

File: src/types/guild.ts [hunk 4]
```diff
@@ -380,6 +381,90 @@ export interface GuildEmbed {
...
+export interface GetGuildAuditLogParams {
+  /** filter the log for actions made by a user */
+  user_id: string;
+  /** the type of audit log event */
+  action_type: AuditLogs;
+  /** filter the log before a certain entry id */
+  before: string;
+  /** how many entries are returned (default 50, minimum 1, maximum 100) */
+  limit: number;
+}
```
Comment (ayntee) [inline src/types/guild.ts:466]: `GetAuditLogsOptions` already exists.

**Comment by ayntee** [comment]
Closing due to inactivity

**Pull request closed without merging**